Core/contentblocker.js

```diff
@@ -18,13 +18,39 @@
 //
 
 (function() {
+    
+    function throttle (callback, limit) {
+        var waiting = false;                      // Initially, we're not waiting
+        return function () {                      // We return a throttled function
+            if (!waiting) {                       // If we're not waiting
+                callback.apply(this, arguments);  // Execute users function
+                waiting = true;                   // Prevent future invocations
+                setTimeout(function () {          // After a period of time
+                    waiting = false;              // And allow future invocations
+                    callback.apply(this, arguments);
+                }, limit);
+            }
+        }
+    }
 
+   let messages = []
+    
+    function sendTrackersDetected() {
+        try {
+            webkit.messageHandlers.trackerDetectedMessage.postMessage(messages);
+        } catch(error) {
+            // webkit might not be defined
+        }
+        messages = []
```

Comment:
There is one more problem with this: it should be cleared on navigation events, otherwise we may be reporting throttled trackers when navigating to a different webpage.